src/anemoi/transform/filters/orog_to_z.py

```diff
@@ -0,0 +1,81 @@
+# (C) Copyright 2024 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from typing import Iterator
+
+import earthkit.data as ekd
+
+from . import filter_registry
+from .matching import MatchingFieldsFilter
+from .matching import matching
+
+
+class Orography(MatchingFieldsFilter):
+    """A filter to convert orography in m to surface geopotential in m²/s², and back."""
+
+    @matching(
+        select="param",
+        forward=("orog"),
+        backward=("z"),
+    )
+    def __init__(
+        self,
+        *,
+        orog="orog",
+        z="z",
+        g=9.80665,
```

Comment:
For constants values like g we should use the constants provided via earthkit meteo. Some examples of those can be found here https://github.com/ecmwf/anemoi-transform/blob/main/src/anemoi/transform/constants.py. So it could be better to remove this as input and use the 'g_gravitational_acceleration' from the constants.py